apps/user_management/middleware.py

```diff
@@ -0,0 +1,14 @@
+from django.utils.deprecation import MiddlewareMixin
+from django.contrib.auth import get_user_model
+
+User = get_user_model()
+
+class GuestSessionMiddleware(MiddlewareMixin):
+    def process_request(self, request):
+        if not request.user.is_authenticated:
+            if 'guest_user_id' in request.session:
+                try:
+                    guest_user = User.objects.get(id=request.session['guest_user_id'])
+                    request.user = guest_user
+                except User.DoesNotExist:
+                    pass
```

Comment:
Shorter and without a try catch block
`.first()` method returns a User object or None otherwise

```python
request.user = User.objects.filter(id=request.session['guest_user_id']).first()
```

Reply:
Also, keep in mind that you may need to add an `is_guest=true` filter